### PR TITLE
use papermill and jupytext for assembly docs

### PR DESCRIPTION
**Pull request by joamatab**
@HelgeGehring

### CODE REVIEW

**Comment by HelgeGehring** [comment]
@joamatab 
https://code.dblock.org/2021/09/03/generating-task-matrix-by-looping-over-repo-files-with-github-actions.html
it seems we can create the matrix automatically from the files in the folder

File: docs/notebooks/elmer_01_electrostatic.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 # ```
 # Afterwards, an easy install method is to add scripts to `~/.local/bin` (or elsewhere in `PATH`) calling the Singularity container for each of the necessary executables. For example, one may create a `ElmerSolver_mpi` file containing
 # ```console
-# #!/bin/bash
+# !/bin/bash
```
Comment (nikosavola) [inline docs/notebooks/elmer_01_electrostatic.py:32]: Does this still render as `#!/bin/bash`?
```suggestion
# #!/bin/bash
```

File: docs/notebooks/palace_01_electrostatic.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 # Alternatively, [Singularity / Apptainer](https://apptainer.org/) containers may be used. Instructions for building and an example definition file are found at [Palace – Build using Singularity/Apptainer](https://awslabs.github.io/palace/dev/install/#Build-using-Singularity/Apptainer).
 # Afterwards, an easy install method is to add a script to `~/.local/bin` (or elsewhere in `PATH`) calling the Singularity container. For example, one may create a `palace` file containing
 # ```console
-# #!/bin/bash
+# !/bin/bash
```
Comment (nikosavola) [inline docs/notebooks/palace_01_electrostatic.py:29]: Same as above
```suggestion
# #!/bin/bash
```

File: docs/notebooks/palace_02_fullwave.py [hunk 3]
```diff
@@ -37,7 +37,7 @@
 # Alternatively, [Singularity / Apptainer](https://apptainer.org/) containers may be used. Instructions for building and an example definition file are found at [Palace – Build using Singularity/Apptainer](https://awslabs.github.io/palace/dev/install/#Build-using-Singularity/Apptainer).
 # Afterwards, an easy install method is to add a script to `~/.local/bin` (or elsewhere in `PATH`) calling the Singularity container. For example, one may create a `palace` file containing
 # ```console
-# #!/bin/bash
+# !/bin/bash
```
Comment (nikosavola) [inline docs/notebooks/palace_02_fullwave.py:40]: Same as above
```suggestion
# #!/bin/bash
```

**Review by nikosavola — CHANGES_REQUESTED** [review]

**Pull request closed without merging**